app/assets/javascripts/discourse/app/components/d-section.gjs

```diff
@@ -0,0 +1,63 @@
+import Component from "@glimmer/component";
+import { scheduleOnce } from "@ember/runloop";
+import notEq from "truth-helpers/helpers/not-eq";
+import deprecated from "discourse-common/lib/deprecated";
+
+// Can add a body class from within a component
+export default class DSection extends Component {
+  <template>
+    {{#if (notEq @tagName "")}}
+      <section id={{@id}} class={{@class}} ...attributes>{{yield}}</section>
+    {{else}}
+      {{yield}}
+    {{/if}}
+  </template>
+
+  currentClasses = new Set();
+
+  constructor() {
+    super(...arguments);
+    scheduleOnce("afterRender", this, this._updateClasses);
+
+    deprecated(
+      `<DSection> is deprecated. Use {{body-class "foo-page" "bar"}} and/or <section></section> instead.`,
+      {
+        since: "3.2.0.beta1",
+        dropFrom: "3.3.0.beta1",
+        id: "discourse.d-section",
+      }
+    );
+  }
+
+  willDestroy() {
+    super.willDestroy(...arguments);
+    scheduleOnce("afterRender", this, this._removeClasses);
+  }
+
+  _updateClasses() {
```

Comment:
How about making d-section lean on the new `{{body-class}}` helper? Something like

```hbs
{{#if @bodyClass}}
  {{body-class @bodyClass}}
{{/if}}

{{#if @pageClass}}
  {{body-class (concat @pageClass "-page")}}
{{/if}}
```
and then we can drop all this custom JS logic